pkg/codegen/go/gen_program.go

```diff
@@ -512,6 +512,9 @@ func (g *generator) getPulumiImport(pkg, vPath, mod, name string) string {
 	if len(modSplit) > 1 {
 		if modSplit[0] == "" || modSplit[0] == "index" {
 			imp = fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s/go/%s", pkg, vPath, pkg)
+			if info.ImportBasePath != "" {
+				imp = info.ImportBasePath
+			}
 		} else {
 			imp = fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s/go/%s/%s", pkg, vPath, pkg, modSplit[0])
 		}
```

Comment:
The `else` block looks wrong here, too.

Reply:
`imp = fmt.Sprintf("%s/%s", info.ImportBasePath, mod)` ?

Reply:
Yeah, or `modSplit[0]`?

Reply:
@AaronFriel actually, we just want this to be `info.ImportBasePath`, if the mod is equal to `index` or empty. Otherwise, we import something like `github.com/muhlba91/pulumi-proxmoxve/sdk/v2/go/proxmoxve/index` unnecessarily.